akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/scaladsl/ActorTestKit.scala

```diff
@@ -173,6 +173,12 @@ final class ActorTestKit private[akka] (val name: String, val config: Config, se
   def spawn[T](behavior: Behavior[T], name: String, props: Props): ActorRef[T] =
     Await.result(internalSystem ? (ActorTestKitGuardian.SpawnActor(name, behavior, _, props)), timeout.duration)
 
+  /**
+   * Stop the actor under test. To make sure it stopped, use [[TestProbe#expectTerminated]] method.
+   */
+  def stop[T](ref: ActorRef[T]): Unit =
+    Await.result(internalSystem ? { x: ActorRef[ActorTestKitGuardian.Ack.type] ⇒ ActorTestKitGuardian.StopActor(ref, x) }, timeout.duration)
```

Comment:
Do we have to `Await.result` here? If the test relies on the actor actually stopping before performing new actions, we could use `TestProbe#expectTerminated` as described in the comment, right?

Reply:
The `Await.result` here merely expects an acknowledgement that the `context.stop()` was called. In the retrospect, not necessary but should work anyway.

Reply:
One idea for a better guarantee would be to watch the actor in the `ActorTestKitGuardian` before stopping it, and not reply here until it has actually stopped. That would make the use cases you mention even simpler as you wouldn't have to also `expectTerminated`.

Reply:
Good idea, but will take more time to implement. I used the existing plumbing to get something workable fast.

Reply:
If the main use case is to run a new actor with the same name directly after - then having to do two separate ops seems strange.

It shouldn't be too hard to implement, in three steps: 
1. add a `case class ActorStopped(replyTo: ActorRef[Ack.type]) extends TestKitCommand` with the other commands
2. where the `ActorTestKitGuardian` has the current stop logic, first do a `context.watch(ref, ActorStopped(replyTo))` before stopping it and remove the part where it sends an ack directly
3. add a case for the `ActorStopped(replyTo)` and send the ack to `replyTo` there instead.

Reply:
Please review the changes. The `stop()` method now also accepts an optional timeout.